SimCalorimetry/EcalSimProducers/src/EcalDigiProducer_Ph2.cc

```diff
@@ -0,0 +1,375 @@
+}
+
+// version for Pre-Mixing, for use outside of MixingModule
+EcalDigiProducer_Ph2::EcalDigiProducer_Ph2(const edm::ParameterSet& params, edm::ConsumesCollector& iC)
+    : DigiAccumulatorMixMod(),
+      m_APDShape(true),
+      m_EBShape(true),
+
+      m_EBdigiCollection(params.getParameter<std::string>("EBdigiCollectionPh2")),
+
+      m_hitsProducerTag(params.getParameter<std::string>("hitsProducer")),
+      m_useLCcorrection(params.getUntrackedParameter<bool>("UseLCcorrection")),
+      m_apdSeparateDigi(params.getParameter<bool>("apdSeparateDigi")),
+
+      m_EBs25notCont(params.getParameter<double>("EBs25notContainment")),
+
+      m_readoutFrameSize(params.getParameter<int>("readoutFrameSize")),
+      m_ParameterMap(new EcalSimParameterMap_Ph2(params.getParameter<double>("simHitToPhotoelectronsBarrel"),
+                                                 params.getParameter<double>("photoelectronsToAnalogBarrel"),
+                                                 params.getParameter<double>("samplingFactor"),
+                                                 params.getParameter<double>("timePhase"),
+                                                 m_readoutFrameSize,
+                                                 params.getParameter<int>("binOfMaximum"),
+                                                 params.getParameter<bool>("doPhotostatistics"),
+                                                 params.getParameter<bool>("syncPhase"))),
+
+      m_apdDigiTag(params.getParameter<std::string>("apdDigiTag")),
+      m_apdParameters(new APDSimParameters(params.getParameter<bool>("apdAddToBarrel"),
+                                           m_apdSeparateDigi,
+                                           params.getParameter<double>("apdSimToPELow"),
+                                           params.getParameter<double>("apdSimToPEHigh"),
+                                           params.getParameter<double>("apdTimeOffset"),
+                                           params.getParameter<double>("apdTimeOffWidth"),
+                                           params.getParameter<bool>("apdDoPEStats"),
+                                           m_apdDigiTag,
+                                           params.getParameter<std::vector<double> >("apdNonlParms"))),
+
+      m_APDResponse(
+          !m_apdSeparateDigi
+              ? nullptr
+              : new EBHitResponse_Ph2(m_ParameterMap.get(), &m_EBShape, true, m_apdParameters.get(), &m_APDShape)),
+
+      m_EBResponse(new EBHitResponse_Ph2(m_ParameterMap.get(),
+                                         &m_EBShape,
+                                         false,  // barrel
+                                         m_apdParameters.get(),
+                                         &m_APDShape)),
+
+      m_PreMix1(params.getParameter<bool>("EcalPreMixStage1")),
+      m_PreMix2(params.getParameter<bool>("EcalPreMixStage2")),
+
+      m_doEB(params.getParameter<bool>("doEB")),
```

Comment:
probably you could remove the doEB parameter and simplify the code accordingly
(since this class only does EB)